enhancements/subscription-content/subscription-injection.md

```diff
@@ -0,0 +1,663 @@
+   default.
+
+**Risk:** Shares and cluster roles are orphaned if the subscription `ClusterBundle` is deleted.
+
+_Mitigation:_ Owner references are used to ensure proper cleanup. Note that `ClusterBundle` is
+cluster-scoped, so it can create owner references on any object.
+
+**Risk:** Containers run with outdated entitlement keys if they are rotated
+
+_Mitigation:_ When an entitlement key is rotated, the secret containing the original key(s) should
+be directly updated with the new entitlement key. This would be treated as a content change by the
+the respective volume mount controller (kubelet for `Bundle`, projected resource csi driver for
+`ClusterBundle`). Updating a `Bundle` or `ClusterBundle` by changing the referenced `entitlements`
+and `yumRepositories` should be clearly discouraged in the documentation.
+
+## Design Details
+
+### Test Plan
+
+This feature set can be tested as follows:
+
+1. Download an OpenShift entitlement key set from the customer portal, and add it to the
+   `openshift-config` namespace as the `etc-pki-entitlement` `Secret`.
+2. Create a `ClusterBundle` object 
+2. Run a pod with the `"subscription.openshift.io/inject-cluster-bundle": cluster` annotation, and
+   a UBI8 container which performs a `yum install` of a RHEL package that requires a subscription.
+
+### Graduation Criteria
+
+This feature will likely require Dev Preview, Tech Preview, and GA maturity levels.
+
+#### Feature Roadmap
+
+1. `Bundle` supporting a single bundle injected into a workload within the same namespace, allowing
+   one entitlement secret and one yum.repo configuration.
+2. `Bundle` supporting multiple `entitlements` and `yumRepositories`.
+3. Allow multiple `Bundles` to be injected into a workload within the same namespace.
+4. Allow a `Bundle` to be automatically injected into a workload within a namespace.
+5. `ClusterBundle` supporting a single bundle injected into a workload, allowing one entitlement
+   secret and one yum.repo configuration.
+6. `ClusterBundle` supporting multiple `entitlements` and `yumRepositories`.
+7. Allow multiple `ClusterBundles` to be injected into a workload.
+8. Allow auto-injection of `ClusterBundles` into all workloads. 
+
+#### Dev Preview
+
+- Bundle APIs at `v1alpha1`.
+- Initial Dev preview release of just the `Bundle` object API and webhook.
+- Subsequent dev preview release of `ClusterBundle` with dependency on the projected resource CSI
+  driver (dev preview).
+
+##### Dev Preview -> Tech Preview
+
+- Projected resource CSI driver reaches tech preview state.
+- APIs reach `v1beta1`
+- Clear documentation on how to enable subscription injection across the cluster
+- Initial scale testing
+- Gather feedback for namespace-restricted subscription bundles
+- Release as a tech preview OLM operator
+
+##### Tech Preview -> GA
+
+- All APIs reach `v1` stability
+- Projected resource CSI driver reaches GA
+- Security audit for these components and the projected resource CSI driver
```

Comment:
@bparees based on the guidelines for security audits, these work best when there is working code to deal with. We have the security audit here as a GA graduation requirement, though there is no reason why we can't ask for one earlier (say after we reach Dev Preview on the csi driver and injection operator).